diff -Naur a/src/dwarf/Gfind_unwind_table.c b/src/dwarf/Gfind_unwind_table.c
--- src/dwarf/Gfind_unwind_table.c	2016-02-22 19:48:57.000000000 +0100
+++ src/dwarf/Gfind_unwind_table.c	2017-10-18 13:41:04.996920564 +0200
@@ -33,6 +33,41 @@
 #include "dwarf-eh.h"
 #include "dwarf_i.h"
 
+static int
+linear_search (unw_addr_space_t as, unw_word_t ip,
+               unw_word_t eh_frame_start, unw_word_t eh_frame_end,
+               unw_word_t fde_count,
+               unw_proc_info_t *pi, int need_unwind_info, void *arg)
+{
+  unw_accessors_t *a = unw_get_accessors (unw_local_addr_space);
+  unw_word_t i = 0, fde_addr, addr = eh_frame_start;
+  int ret;
+
+  while (i++ < fde_count && addr < eh_frame_end)
+    {
+      fde_addr = addr;
+      if ((ret = dwarf_extract_proc_info_from_fde (as, a, &addr, pi,
+                                                   eh_frame_start,
+                                                   0, 0, arg)) < 0)
+        return ret;
+
+      if (ip >= pi->start_ip && ip < pi->end_ip)
+        {
+          if (!need_unwind_info)
+            return 1;
+          addr = fde_addr;
+          if ((ret = dwarf_extract_proc_info_from_fde (as, a, &addr, pi,
+                                                       eh_frame_start,
+                                                       need_unwind_info, 0,
+                                                       arg))
+              < 0)
+            return ret;
+          return 1;
+        }
+    }
+  return -UNW_ENOINFO;
+}
+
 int
 dwarf_find_unwind_table (struct elf_dyn_info *edi, unw_addr_space_t as,
                          char *path, unw_word_t segbase, unw_word_t mapoff,
@@ -160,7 +195,7 @@
 
       if (hdr->table_enc != (DW_EH_PE_datarel | DW_EH_PE_sdata4))
         {
-    #if 1
+    #if 0
           abort ();
     #else
           unw_word_t eh_frame_end;
@@ -182,7 +217,7 @@
 
           return linear_search (unw_local_addr_space, ip,
                                 eh_frame_start, eh_frame_end, fde_count,
-                                pi, need_unwind_info, NULL);
+                                &pi, /*need_unwind_info*/0, NULL);
     #endif
         }
 
