diff -ur include/libunwind-coredump.h include/libunwind-coredump.h
--- include/libunwind-coredump.h	2012-10-06 06:54:38.000000000 +0200
+++ include/libunwind-coredump.h	2014-07-08 09:42:09.996805450 +0200
@@ -37,6 +37,14 @@
 
 struct UCD_info;
 
+typedef struct
+  {
+	uint64_t startAddr;
+	uint64_t endAddr;
+	uint64_t pageOffset;
+	const char *filename; /* points inside note_phdr */
+  } file_t;
+
 extern struct UCD_info *_UCD_create(const char *filename);
 extern void _UCD_destroy(struct UCD_info *);
 
@@ -48,6 +56,7 @@
 extern int _UCD_add_backing_file_at_vaddr(struct UCD_info *,
 				          unsigned long vaddr,
 					  const char *filename);
+extern int _UCD_add_backing_files_from_file_note(struct UCD_info *);
 
 extern int _UCD_find_proc_info (unw_addr_space_t, unw_word_t,
 				unw_proc_info_t *, int, void *);
@@ -65,6 +74,8 @@
 extern int _UCD_resume (unw_addr_space_t, unw_cursor_t *, void *);
 extern unw_accessors_t _UCD_accessors;
 
+extern unsigned _UCD_get_num_note_files(struct UCD_info *ui);
+extern file_t *_UCD_get_num_note_file(struct UCD_info *ui, unsigned n);
 
 #if defined(__cplusplus) || defined(c_plusplus)
 }
diff -ur src/coredump/_UCD_create.c src/coredump/_UCD_create.c
--- src/coredump/_UCD_create.c	2012-10-06 06:54:38.000000000 +0200
+++ src/coredump/_UCD_create.c	2014-07-08 09:41:49.212457968 +0200
@@ -63,6 +63,10 @@
 #include <elf.h>
 #include <sys/procfs.h> /* struct elf_prstatus */
 
+#ifndef NT_FILE         /* older kernel versions do yet have this note */
+#define NT_FILE         0x46494c45
+#endif
+
 #include "_UCD_lib.h"
 #include "_UCD_internal.h"
 
@@ -246,6 +250,27 @@
                 if (note_hdr->n_type == NT_PRSTATUS)
                   ui->threads[n_threads++] = NOTE_DATA (note_hdr);
 
+                if (note_hdr->n_type == NT_FILE)
+                  {
+					char *filedata = NOTE_DATA(note_hdr);
+#define GET_AND_INCREMENT(ptr) (_64bits ? *((uint64_t*)ptr) : *((uint32_t*)ptr)); ptr += _64bits ? 8 : 4;
+					ui->files_count = GET_AND_INCREMENT(filedata);
+					uint64_t pageSize = GET_AND_INCREMENT(filedata);
+					(void) pageSize;
+					ui->files = (file_t *) malloc(sizeof(file_t)*ui->files_count);
+					for (i = 0; i < ui->files_count; i++)
+					  {
+						ui->files[i].startAddr = GET_AND_INCREMENT(filedata);
+						ui->files[i].endAddr = GET_AND_INCREMENT(filedata);
+						ui->files[i].pageOffset = GET_AND_INCREMENT(filedata);
+#undef GET_AND_INCREMENT
+					  }
+					for (i = 0; i < ui->files_count; i++)
+					  {
+						ui->files[i].filename = filedata;
+						filedata += strlen(filedata)+1;
+					  }
+                  }
                 note_hdr = NOTE_NEXT (note_hdr);
               }
           }
@@ -415,3 +440,29 @@
     }
   return -1;
 }
+
+int _UCD_add_backing_files_from_file_note(struct UCD_info *ui)
+{
+  if (ui->files)
+    {
+	  unsigned i;
+	  for (i = 0; i < ui->files_count; i++)
+	    {
+		  if (ui->files[i].pageOffset == 0) _UCD_add_backing_file_at_vaddr(ui, ui->files[i].startAddr, ui->files[i].filename);
+	    }
+
+    }
+  return -1;
+}
+
+unsigned _UCD_get_num_note_files(struct UCD_info *ui)
+{
+  return ui->files_count;
+}
+
+file_t *_UCD_get_num_note_file(struct UCD_info *ui, unsigned n)
+{
+  if (n < ui->files_count)
+    return &ui->files[n];
+  return NULL;
+}
diff -ur src/coredump/_UCD_destroy.c src/coredump/_UCD_destroy.c
--- src/coredump/_UCD_destroy.c	2012-10-06 06:54:38.000000000 +0200
+++ src/coredump/_UCD_destroy.c	2014-07-08 09:38:34.809198542 +0200
@@ -44,6 +44,7 @@
         close(phdr->backing_fd);
     }
 
+  free(ui->files);
   free(ui->note_phdr);
 
   free(ui);
diff -ur src/coredump/_UCD_internal.h src/coredump/_UCD_internal.h
--- src/coredump/_UCD_internal.h	2012-10-06 06:54:38.000000000 +0200
+++ src/coredump/_UCD_internal.h	2014-07-07 11:28:03.000000000 +0200
@@ -91,6 +91,8 @@
     unsigned phdrs_count;
     void *note_phdr; /* allocated or NULL */
     struct PRSTATUS_STRUCT *prstatus; /* points inside note_phdr */
+    file_t *files; /* allocated or NULL */
+    unsigned files_count;
     int n_threads;
     struct PRSTATUS_STRUCT **threads;
 
